src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/feature/OcrDataOrderTest.java

```diff
@@ -0,0 +1,60 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.feature;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.SampleData;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.helper.CcdCollectionElementComparator;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.model.ccd.CcdCollectionElement;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.model.ccd.CcdKeyValue;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.model.ccd.ExceptionRecord;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.model.ccd.mappers.ExceptionRecordMapper;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.EnvelopeParser;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.model.Envelope;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class OcrDataOrderTest {
+
+    @DisplayName("Should parse incoming envelope with OCR data and map to same order in CCD record")
+    @Test
+    public void sameOrder() {
+        // given
+        byte[] envelopeMessage = SampleData.exampleJsonAsBytes();
+
+        // when
+        Envelope envelope = EnvelopeParser.parse(envelopeMessage);
+        assertThat(envelope.ocrData).isInstanceOf(LinkedHashMap.class);
+
+        // and
+        ExceptionRecordMapper mapper = new ExceptionRecordMapper();
+        ExceptionRecord record = mapper.mapEnvelope(envelope);
+        assertThat(record.ocrData).isInstanceOf(ArrayList.class);
+        assertThat(record.ocrData.size()).isEqualTo(envelope.ocrData.size());
+
+        // then
+        int i = 0;
+        Iterator<Map.Entry<String, String>> entries = envelope.ocrData.entrySet().iterator();
+
+        while (entries.hasNext()) {
+            Map.Entry<String, String> expectedEntry = entries.next();
+            assertThat(record.ocrData.get(i).value.key).isEqualTo(expectedEntry.getKey());
```

Comment:
do we need these asserts ? We already have below asserts to validate they are in same order ?

Reply:
I would prefer just keeping a loop which assures any size of ocr data order. i think i included the below section for giggles. for people who prefer reading text rather than code